main.go

```diff
@@ -172,6 +173,31 @@ func configureEnvironment() (dataSource, port string, migrationSource string, ma
 	return dataSource, port, migrationSource, maxIdleConn, connMaxLifetime
 }
 
+func handleShutdownSignals(server *http.Server, timeout time.Duration) {
+	defer os.Exit(0)
+
+	stopSignals := make(chan os.Signal, 1)
+	signal.Notify(stopSignals, syscall.SIGINT, syscall.SIGTERM)
+	stopSignal := <-stopSignals
```

Comment:
Is this essentially waiting until until signal.Notify puts a signal in the channel?

Reply:
Yes, it blocks the rest of the function until it receives something on that channel.